src/main/java/org/opensearch/flowframework/workflow/GetResourceStep.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.support.PlainActionFuture;
+import org.opensearch.flowframework.model.ResourceCreated;
+
+import java.util.Map;
+
+/**
+ * Internal step to pass created resources to dependent nodes. Only used in reprovisioning
+ */
+public class GetResourceStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(GetResourceStep.class);
+    private final ResourceCreated resourceCreated;
+
+    /** The name of this step */
+    public static final String NAME = "get_resource";
+
+    /**
+     * Instantiate this class
+     * @param resourceCreated the created resource
+     */
+    public GetResourceStep(ResourceCreated resourceCreated) {
```

Comment:
Continuing on the naming thing, we're not really "getting" the resource; we must already have it.  We're just outputting the given resource.

This whole step/class could probably be made very generic, call it a `WorkflowDataStep` and just pass it a map argument and insert it into the workflow data.

Reply:
Ill change the name